nuitka/build/Backend.scons

```diff
@@ -825,7 +829,17 @@ def getWindowsPythonDLLPath():
 
 if win_target:
     # Some non standard Pythons on Windows have this.
-    if gcc_mode and static_libpython:
+    if sys.version_info[0] > 2 and sys.implementation.name == "nuitkapython":
```

Comment:
The Python3 check seems redundent, also need to check `python_version` as passed from the outside, as Scons itself may be running with another Python, although that's going to be rare, but I would just remove the check here.

Reply:
The reason the check is there is that  sys.implementation.name does not exist on 2.7 and it caused the CI to fail. So I will just change the check to use python_version